app/models/item.rb

```diff
@@ -1,10 +1,34 @@
 class Item < ActiveRecord::Base
-  validates :name, :description, :price, :seller_id, presence: true
+  validates :name, :description, :seller_id, presence: true
+  validate :price, :auction_details, :auction_duration
 
   scope :available, -> { where(available: true) }
 
   belongs_to :seller, class_name: "User", foreign_key: :seller_id
   has_one :purchaser, through: :purchases, source: :purchaser
+  has_many :bids
 
   mount_uploader :image, ImageUploader
+
+  def highest_bid
+    bids.maximum(:amount) || starting_bid_price
+  end
+
+  def price
+    if !buy_it_now_price && !starting_bid_price
+      errors.add(:price, "you need to select either a starting bid or a buy it now price")
+    end
+  end
+
+  def auction_details
+    if (starting_bid_price && !auction_end_time) || (!starting_bid_price && auction_end_time)
+      errors.add(:auction_details, "you must select a starting_bid and auction end time")
+    end
+  end
+
+  def auction_duration
+    if auction_end_time && (auction_end_time - Time.now) <= 3600
+      errors.add(:auction_duration, "you cannot schedule an auction to be under an hour")
```

Comment:
We made this decision so users don't leave the default time for the auction end field because the bids would expire instantly